lib/charms/opensearch/v0/opensearch_snapshots.py

```diff
@@ -0,0 +1,1628 @@
+                    "s3.client.default.secret_key": info["secret_key"],
+                }
+            )
+            self.charm.keystore_manager.reload()
+
+            # Optional CA chain
+            if info.get("s3_tls_ca_chain"):
+                logger.info("S3 TLS CA Chain detected.")
+                self.charm.snapshots_manager.store_s3_ca(info["s3_tls_ca_chain"])
+                self._restart_for_ca(info["s3_tls_ca_chain"], reason="apply new object storage CA")
+
+            else:
+                if self.charm.snapshots_manager.is_custom_s3_ca_stored():
+                    self.charm.snapshots_manager.remove_s3_ca()
+                    self._restart_for_ca(None, reason="clean up the object storage CA")
+
+        elif event.kind == "azure":
+            info = self._read_azure_from_peer()
+            if not (info and info["storage_account"] and info["secret_key"]):
+                logger.warning("No Azure storage configuration.")
+                return
+
+            self.charm.keystore_manager.put_entries(
+                {
+                    "azure.client.default.account": info["storage_account"],
+                    "azure.client.default.key": info["secret_key"],
+                }
+            )
+            logger.info("Azure storage credentials are added to keystore.")
+            self.charm.keystore_manager.reload()
+
+    def _on_object_storage_gone_on_peers(self, event: _ObjectStorageGone) -> None:
+        if event.kind == "s3":
+            if self._cleanup(
+                "s3",
+                ["s3.client.default.access_key", "s3.client.default.secret_key"],
+            ):
+                if self.charm.snapshots_manager.is_custom_s3_ca_stored():
+                    self.charm.snapshots_manager.remove_s3_ca()
+                    self._restart_for_ca(None, reason="clean up the object storage CA")
+        elif event.kind == "azure":
+            self._cleanup("azure", ["azure.client.default.account", "azure.client.default.key"])
+
+    def _current_s3_ca_chain(self) -> str:
+        """Return the currently stored S3 CA chain (string) from cacerts, or ''."""
+        try:
+            stored = (
+                list_cas(
+                    # TODO: create a S3 CA truststore juju secret
+                    #  and it should be known by subclusters
+                    store_pwd="changeit",
+                    store_path=f"{self.charm.opensearch.paths.certs}/cacerts.p12",
+                )
+                or {}
+            )
+            return stored.get(S3_CA_ALIAS) or ""
+        except Exception:
+            return ""
+
+    def _ca_changed(self, new_chain: str | None) -> bool:
+        current_norm = _normalize_chain(self._current_s3_ca_chain())
+        new_norm = _normalize_chain(new_chain)
+        return _hash(current_norm) != _hash(new_norm)
+
+    def _restart_for_ca(self, new_chain: str | None, *, reason: str) -> bool:
```

Comment:
nit, perhaps for clarity
```suggestion
    def _restart_for_ca_if_needed(self, new_chain: str | None, *, reason: str) -> bool:
```

Reply:
Done.